### DISCUSSION TITLE
Idea on managing IDs for books and maybe authors

### DISCUSSION THREAD

**Discussion by jaschaurbach** [bookwyrm-social/bookwyrm#2521] [Ideas]
I was thinking about the wealth of IDs for Books and authors and the following came to my mind:

Different instances have different needs on IDs, even on an instance user may have different needs. My idea works on instance level, maybe on a user level. 

At the moment the list of IDs its growing and I alone have so many ideas which IDs could be used to identify books and I am sure folks have even more ideas for IDs. The list is growing and growing and will become overwhelming for users.

I propose that on instance level admins get a control panel to decide which IDs will be shown to users and will be editable from the standard book edit view. Admins can decide if there will be a button "more IDs" which will open up a modal to show all IDs or if this button will not be shown and make the IDs inaccessible for users. 

For a book view this will change as well which links are shown on the left side below the cover. Either these links are not shown at all or for that there is a admin control as well. 

Same goes for the edition overview: instance admins should be able to decide which ID will be shown or not. 

Regardless of the setting, all IDs will be synced into the database - if an admin decides later to change which IDs should be shown they are present. 

The same technique can be used for authors.

Example: I am running a more SciFi/Fantasy oriented  instance. The ISFDB.org ID is for this instance important. At an instance not focussing on this topic, the ID may be more confusing than that. 

There should be a preset which can not be deactivated to be shown to users that there is a minimum set that all users on an instance can expect to be present.

**Comment by chdorner** [comment]
This could be a good place to give more context to admins about what some of the more specific IDs are useful for. For example I didn't know that ISFDB.org is useful for SciFi/Fantasy.

**Reply by jaschaurbach** [reply]
I just began to write down about the IDs for the documentation. I think this is the right place for explanation of IDs, as they are relevant for users as well. Maybe we could link to the documentation once it is finished?
